utils/node_local_test.py

```diff
@@ -1117,6 +1140,70 @@ def run_daos_client_cmd_pil4dfs(self, cmd, check=True, container=None):
         print(f'DBG> num_op = {num_op}')
         return rc
 
+    def set_fi(self, probability=0):
+        """Run the client code to set server params"""
+        cmd_env = get_base_env()
+
+        cmd_env['OFI_INTERFACE'] = self.network_interface
+        cmd_env['CRT_PHY_ADDR_STR'] = self.network_provider
+        valgrind_hdl = ValgrindHelper(self.conf)
+
+        if self.conf.args.memcheck == 'no':
+            valgrind_hdl.use_valgrind = False
+
+        system_name = 'daos_server'
+
+        exec_cmd = valgrind_hdl.get_cmd_prefix()
+
+        agent_bin = join(self.conf['PREFIX'], 'bin', 'daos_agent')
+
+        with tempfile.TemporaryDirectory(prefix='dnt_addr_',) as addr_dir:
+
+            addr_file = join(addr_dir, f'{system_name}.attach_info_tmp')
+
+            agent_cmd = [agent_bin,
+                         '-i',
+                         '-s',
+                         self.agent_dir,
+                         'dump-attachinfo',
+                         '-o',
+                         addr_file]
+
+            rc = subprocess.run(agent_cmd, env=cmd_env, check=True)
+            print(rc)
+
+            # options here are: fault_id,max_faults,probability,err_code[,argument]
+            cmd = ['set_fi_attr',
+                   '--cfg_path',
+                   addr_dir,
+                   '--group-name',
+                   'daos_server',
+                   '--rank',
+                   '0',
+                   '--attr',
+                   f'0,0,{probability},0,0']
+
+            exec_cmd.append(join(self.conf['PREFIX'], 'bin', 'cart_ctl'))
+            exec_cmd.extend(cmd)
+
+            with tempfile.NamedTemporaryFile(prefix=f'dnt_crt_ctl_{get_inc_id()}_',
+                                             suffix='.log',
+                                             delete=False) as log_file:
+
+                cmd_env['D_LOG_FILE'] = log_file.name
+                cmd_env['DAOS_AGENT_DRPC_DIR'] = self.agent_dir
+
+                rc = subprocess.run(exec_cmd,
+                                    env=cmd_env,
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE,
+                                    check=False)
+                print(rc)
+                valgrind_hdl.convert_xml()
+                log_test(self.conf, log_file.name)
+                # Do not check returncode here, fault injection might prevent it from working.
+                return False  # fatal_errors
```

Comment:
This could probably be removed, I'll check.  I'm not sure anything checks the return code here and the comment is wrong, I'm also surprised pylint allows it.